### PR TITLE
[Merged by Bors] - refactor: deprecate `ContinuousMonoidHomClass`

### PR DESCRIPTION
**Pull request by urkud**
Use `[ContinuousMapClass ..] [MonoidHomClass ..]` instead. Also deprecate `mk'` since it's defeq `mk`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by YaelDillies — APPROVED** [review]
Now that `FunLike` is not extended, this makes a lot of sense.

maintainer merge

File: Mathlib/Topology/Algebra/ContinuousMonoidHom.lean [hunk 1]
```diff
@@ -30,49 +30,43 @@ variable (F A B C D E : Type*) [Monoid A] [Monoid B] [Monoid C] [Monoid D] [Comm
 /-- The type of continuous additive monoid homomorphisms from `A` to `B`.
 
 When possible, instead of parametrizing results over `(f : ContinuousAddMonoidHom A B)`,
-you should parametrize over `(F : Type*) [ContinuousAddMonoidHomClass F A B] (f : F)`.
+you should parametrize
+over `(F : Type*) [FunLike F A B] [ContinuousMapClass F A B] [AddMonoidHomClass F A B] (f : F)`.
 
 When you extend this structure, make sure to extend `ContinuousAddMonoidHomClass`. -/
```
Comment (sgouezel) [inline Mathlib/Topology/Algebra/ContinuousMonoidHom.lean:36]: Can you adapt the docstring to the new setting?

File: Mathlib/Topology/Algebra/ContinuousMonoidHom.lean [hunk 2]
```diff
@@ -82,66 +76,65 @@ add_decl_doc ContinuousMonoidHom.toMonoidHom
...
 @[to_additive]
 theorem toContinuousMap_injective : Injective (toContinuousMap : _ → C(A, B)) := fun f g h =>
   ext <| by convert DFunLike.ext_iff.1 h
 
 -- Porting note: Removed simps because given definition is not a constructor application
 /-- Construct a `ContinuousMonoidHom` from a `Continuous` `MonoidHom`. -/
-@[to_additive "Construct a `ContinuousAddMonoidHom` from a `Continuous` `AddMonoidHom`."]
+@[to_additive (attr := deprecated (since := "2024-10-08"))
+  "Construct a `ContinuousAddMonoidHom` from a `Continuous` `AddMonoidHom`."]
 def mk' (f : A →* B) (hf : Continuous f) : ContinuousMonoidHom A B :=
```
Comment (sgouezel) [inline Mathlib/Topology/Algebra/ContinuousMonoidHom.lean:118]: If it's defeq to something already existing, could you rather write it as a deprecated alias, to make sure that people who were using it get the suggestion to use `mk` instead?

**Review by sgouezel — COMMENTED** [review]
bors d+
Thanks!

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**